microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/core/StaticProviderRegistry.java

```diff
@@ -0,0 +1,40 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+* which is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lsp4mp.jdt.internal.core;
+
+import org.eclipse.lsp4mp.jdt.core.IPropertiesProvider;
+
+/**
+ * Registry to hold the Extension point
+ * "org.eclipse.lsp4mp.jdt.core.staticProviders".
+ *
+ */
+public class StaticProviderRegistry extends AbstractMicroProfileStaticProviderRegistry<IPropertiesProvider> {
+
+    private static final StaticProviderRegistry INSTANCE = new StaticProviderRegistry();
+
+    public static StaticProviderRegistry getInstance() {
+        return INSTANCE;
+    }
+
+    private StaticProviderRegistry() {
+        super();
+    }
+
+    @Override
+    public String getProviderExtensionId() {
+        return "staticProviders";
```

Comment:
I think it makes more sense to call this extension point `staticPropertyProviders`, just to be extra clear.